File: readability.py
from cs50 import get_string

def main():
    # Declare variables (don't need to specify type)
    # letters = 0
    # words = 0
    # sentences = 0

    # Prompt user for text
    text = get_string("Text: ")

    # Calculate number of letters, words, and sentences
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    print(str(letters) + " " + str(words) + " " + str(sentences))

    # Coleman-Liau formula
    L = letters / words * 100
    S = sentences / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8

    # Print reading level
    if (index >= 1 and index < 16):
        print(f"Grade {int(round(index))}")
    elif (index < 1):
        print("Before Grade 1")
    else:
        print("Grade 16+")

def count_letters(text):
    letters = 0
    for i in text:
        # if string.isalpha(text(i)):
        if i.isalpha():
            letters += 1
    return letters

def count_words(text):
    words = 1
    for i in text:
        # if string.isspace(text(i)):
        if i.isspace():
            words += 1
    return words

def count_sentences(text):
    sentences = 0
    for i in text:
        if (i == "." or i == "!" or i == "?"):
            sentences += 1
    return sentences

if __name__ == "__main__":
    main()

File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

//Declare variables.
float letters;
float words = 1;
float sentences;

//Prototype functions.
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int main(void)
{
    //Prompt user for text.
    string text = get_string("Text: ");
    //printf("%s\n", text);
    //Calculate number of letters, words, and sentences.
    count_letters(text);
    //printf("%f letters\n", letters);
    count_words(text);
    //printf("%f words\n", words);
    count_sentences(text);
    //printf("%f sentences\n", sentences);
    //Coleman-Liau formula.
    float L = letters / words * 100;
    //printf("L is %f\n", L);
    float S = sentences / words * 100;
    //printf("S is %f\n", S);
    float index = 0.0588 * L - 0.296 * S - 15.8;
    //printf("Index is %f\n", index);
    //Display reading level.
    if (index >= 1 && index < 16)
    {
        printf("Grade %i\n", (int) round(index));
    }
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
}

//Function definitions for counting letters, words, and sentences.
int count_letters(string text)
{
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if isalpha(text[i])
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if isspace(text[i])
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            sentences++;
        }
    }
    return sentences;
}